src/lib/pickles_base/domain.ml

```diff
@@ -1,5 +1,7 @@
 open Core_kernel
 
+[@@@warning "-4"] (* sexp-related fragile pattern-matching warning *)
```

Comment:
We might want to consider upstreaming a fix for this at some point, if there's an obvious one. (That fix might turn out to be localising this to the ppx, I suppose.)

Reply:
I haven't yet looked at the generated code but I am not a big fan of how this is fixed in our codebase as of right now.